test-runner/tests/test_test_suite_runner.py

```diff
@@ -62,8 +62,12 @@ def test_runner_end_to_end() -> None:
 
     reporters = [Mock(), Mock()]
 
+    filt = Mock()
+    filt.should_skip.return_value = (False, None)
+    filters = [filt]
+
     with patch("glob.glob", return_value=test_paths):
-        suite = tsr.run_tests_from_test_suite("my-path", runtime, validators, reporters)  # type: ignore
+        suite = tsr.run_tests_from_test_suite("my-path", runtime, validators, reporters, filters)  # type: ignore
```

Comment:
should we validate if filters actually were called?

Reply:
done

Reply:
it'd be also good to validate the case when the filter returns `False` and in that case the test is skipped